layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayout.kt

```diff
@@ -438,4 +430,61 @@ open class FoldableLayout @JvmOverloads constructor(
 
         var isDualLandscapeSingleContainer: Boolean = false
     )
+
+    fun newConfigCreator() = BaseConfig.NewConfigCreator(this)
+    fun updateConfigCreator() = BaseConfig.UpdateConfigCreator(this)
+
+    /**
+     * Base class to keep the configuration of the SurfaceDuoLayout
+     *
+     * @param <T> An object that extends BaseConfig
+     */
+    @Suppress("UNCHECKED_CAST")
+    sealed class BaseConfig<T : BaseConfig<T>>(protected val config: Config) {
+
+        /**
+         * Class to add a new config in SurfaceDuoLayout
+         * and recreate the view
+         */
+        class NewConfigCreator(
+            private val surfaceDuoLayout: FoldableLayout
```

Comment:
Please, change the name of the instance to `foldableLayout`.

Reply:
Done